pypose/module/dynamics.py

```diff
@@ -0,0 +1,229 @@
+from numpy import vectorize
+import torch as torch
+import torch.nn as nn
+from torch.autograd.functional import jacobian
+
+class _System(nn.Module):
+    r'''
+    A sub-class of :obj:`torch.nn.Module` to build general dynamics.
+    
+    Args:
+        time (:obj:`boolean`): Whether the system is time-varying; defaults to False, meaning time-invariant
+    '''
+    def __init__(self, time=False):
+        super().__init__()
+        self.jacargs = {'vectorize':True, 'strategy':'reverse-mode'}
+        if time:
+            self.register_buffer('t',torch.zeros(1))
+            self.register_forward_hook(self.forward_hook)
+
+    def forward_hook(self, module, inputs, outputs):
+        self.input, self.state = inputs
+        self.t.add_(1)
+
+    def forward(self, state, input):
+        r'''
+        Parameters
+        ----------
+        state : Tensor
+                The state of the dynamic system
+        input : Tensor
+                The input to the dynamic system
+
+        Returns
+        -------
+        new_state   : Tensor
+                      The state of the system at next time step
+        observation : Tensor
+                      The observation of the system at the current step
+        '''
+
+        new_state = self.state_transition(state, input)
+        observation = self.observation(state, input)
+        return new_state, observation
+
+    def state_transition(self):
+        pass
+

```

Comment:
Better do
def state_transition(self, state, input):
    raise NotImplementedError("The users need to define their own state transition method")
Same for observation.

Also, add the documentation too.

Reply:
The LTI system does not define a state_transition or observation function. It will throw an error when that class is used. Should I still implement this?